feature_engine/timeseries/forecasting/window_features.py

```diff
@@ -0,0 +1,333 @@
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class WindowFeatures(BaseEstimator, TransformerMixin):
+    """
+        LagFeatures adds lag features to the dataframe. A lag feature is a feature with
+        information about a prior time step.
+
+        LagFeatures has the same functionality as pandas `shift()` with the exception that
+        only one of `periods` or `freq` can be indicated at a time. LagFeatures builds on
+        top of pandas `shift()` in that multiple lags can be created at the same time and
+        the features with names will be concatenated to the original dataframe.
+
+        To be compatible with LagFeatures, the dataframe's index must have unique values
+        and no NaN.
+
+        LagFeatures works only with numerical variables. You can pass a list of variables
+        to lag. Alternatively, LagFeatures will automatically select and lag all numerical
+        variables found in the training set.
+
+        More details in the :ref:`User Guide <lag_features>`.
+
+        Parameters
+        ----------
+        {variables}
+
+        periods: int, list of ints, default=1
+            Number of periods to shift. Can be a positive integer or list of positive
+            integers. If list, features will be created for each one of the periods in the
+            list. If the parameter `freq` is specified, `periods` will be ignored.
+
+        freq: str, list of str, default=None
+            Offset to use from the tseries module or time rule. See parameter `freq` in
+            pandas `shift()`. It is the same functionality. If freq is a list, lag features
+            will be created for each one of the frequency values in the list. If freq is not
+            None, then this parameter overrides the parameter `periods`.
+
+        sort_index: bool, default=True
+            Whether to order the index of the dataframe before creating the lag features.
+
+        {missing_values}
+
+        {drop_original}
+
+        Attributes
+        ----------
+        variables_:
+            The group of variables that will be lagged.
+
+        {feature_names_in_}
+
+        {n_features_in_}
+
+        Methods
+        -------
+        {fit}
+
+        transform:
+            Add lag features.
+
+        {fit_transform}
+
+        See Also
+        --------
+        pandas.shift
+        """
+
+    def __init__(
+            self,
+            variables: List[str] = None,
+            window: Union[str, int] = 1,
+            function: Callable = np.mean,
+            periods: int = 1,
+            freq: str = None,
+            sort_index: bool = True,
+            missing_values: str = "raise",
+            drop_original: bool = False,
+    ) -> None:
+
+        if not (isinstance(window, int) or isinstance(window, str)) or isinstance(window, bool):
```

Comment:
Like in my previous comment, I would leave all checks of parameters that will be used by pandas, to pandas. That is, periods, freq and window.